moveit_ros/perception/pointcloud_octomap_updater/src/pointcloud_octomap_updater.cpp

```diff
@@ -66,61 +66,53 @@ PointCloudOctomapUpdater::~PointCloudOctomapUpdater()
   stopHelper();
 }
 
-bool PointCloudOctomapUpdater::setParams(XmlRpc::XmlRpcValue& params)
+bool PointCloudOctomapUpdater::setParams(const std::string& name_space)
 {
-  try
-  {
-    if (!params.hasMember("point_cloud_topic"))
-      return false;
-    point_cloud_topic_ = static_cast<const std::string&>(params["point_cloud_topic"]);
-
-    readXmlParam(params, "max_range", &max_range_);
-    readXmlParam(params, "padding_offset", &padding_);
-    readXmlParam(params, "padding_scale", &scale_);
-    readXmlParam(params, "point_subsample", &point_subsample_);
-    if (params.hasMember("max_update_rate"))
-      readXmlParam(params, "max_update_rate", &max_update_rate_);
-    if (params.hasMember("filtered_cloud_topic"))
-      filtered_cloud_topic_ = static_cast<const std::string&>(params["filtered_cloud_topic"]);
-  }
-  catch (XmlRpc::XmlRpcException& ex)
-  {
-    ROS_ERROR_STREAM_NAMED(LOGNAME, "XmlRpc Exception: " << ex.getMessage());
-    return false;
-  }
-
-  return true;
+  return node_->get_parameter(name_space + ".point_cloud_topic", point_cloud_topic_) &&
+         node_->get_parameter(name_space + ".max_range", max_range_) &&
+         node_->get_parameter(name_space + ".padding_offset", padding_) &&
+         node_->get_parameter(name_space + ".padding_scale", scale_) &&
+         node_->get_parameter(name_space + ".point_subsample", point_subsample_) &&
+         node_->get_parameter(name_space + ".max_update_rate", max_update_rate_) &&
+         node_->get_parameter(name_space + ".filtered_cloud_topic", filtered_cloud_topic_);
 }
 
-bool PointCloudOctomapUpdater::initialize()
+bool PointCloudOctomapUpdater::initialize(const rclcpp::Node::SharedPtr node)
 {
-  tf_buffer_.reset(new tf2_ros::Buffer());
-  tf_listener_.reset(new tf2_ros::TransformListener(*tf_buffer_, root_nh_));
+  node_ = node;
+  rclcpp::Clock::SharedPtr clock = std::make_shared<rclcpp::Clock>(RCL_SYSTEM_TIME);
```

Comment:
Does the TFBuffer require system time?

Reply:
You are correct. I think the TFBuffer here doesn't need to use system time.